src/components/shared/modals/modalWithInput/index.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {Dispatch, SetStateAction, useEffect} from 'react';
+import {Button, Modal, Text} from 'native-base';
+import Input from 'components/shared/input';
+import styles from './styles';
+import {useForm} from 'react-hook-form';
+import {schema} from './validations';
+import {yupResolver} from '@hookform/resolvers/yup';
+import {Keyboard} from 'react-native';
+
+interface Props {
+  buttonText: string;
+  headerText: string;
+  firstInputName: string;
+  isVisible: boolean;
+  setIsVisible: Dispatch<SetStateAction<boolean>>;
+  secondInputName?: string;
+  onSubmit: (data: {[x: string]: string}) => void;
+}
+
+const ModalWithInput = ({
+  headerText,
+  firstInputName,
+  secondInputName,
+  onSubmit,
+  isVisible,
+  setIsVisible,
+}: Props) => {
+  const defaultValues = {
+    [firstInputName]: '',
+    [secondInputName as string]: '',
+  };
+  const {
+    control,
+    formState: {errors},
+    setError,
+    clearErrors,
+    handleSubmit,
+    setValue,
+  } = useForm({
+    defaultValues: defaultValues,
+    resolver: yupResolver(schema),
+    mode: 'onSubmit',
+    reValidateMode: 'onBlur',
+  });
+
+  const onPress = (data: {[x: string]: string}) => {
+    if (!data.firstInputValue || !data.secondInputValue) {
+      Keyboard.dismiss();
+      setError(firstInputName, {
+        type: 'Invalid email or password',
+        message: 'You need to enter a valid user and password',
+      });
+    }
+    onSubmit(data);
+  };
+
+  useEffect(() => {
+    return () => {
+      setValue(firstInputName, '');
+      setValue(secondInputName as string, '');
+    };
+  }, [firstInputName, isVisible, secondInputName, setValue]);
```

Comment:
This is not the right way to reset the form, as a modal is not unmounted when closed, it just becomes 'invisible'. Also, the right way to reset a form in react-hook-form is to use the reset method
```suggestion

  useEffect(() => {
    if(!isVisible) reset()
  }, [reset, isVisible]);
```

Reply:
done